crates/accelerate/src/basis/basis_translator/basis_search.rs

```diff
@@ -0,0 +1,228 @@
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use std::cell::RefCell;
+
+use hashbrown::{HashMap, HashSet};
+use pyo3::prelude::*;
+
+use crate::equivalence::{EdgeData, Equivalence, EquivalenceLibrary, Key, NodeData};
+use qiskit_circuit::operations::Operation;
+use rustworkx_core::petgraph::stable_graph::{EdgeReference, NodeIndex, StableDiGraph};
+use rustworkx_core::petgraph::visit::Control;
+use rustworkx_core::traversal::{dijkstra_search, DijkstraEvent};
+
+use super::compose_transforms::{BasisTransformIn, GateIdentifier};
+
+#[pyfunction]
+#[pyo3(name = "basis_search")]
+/// Search for a set of transformations from source_basis to target_basis.
+/// Args:
+///     equiv_lib (EquivalenceLibrary): Source of valid translations
+///     source_basis (Set[Tuple[gate_name: str, gate_num_qubits: int]]): Starting basis.
+///     target_basis (Set[gate_name: str]): Target basis.
+///
+/// Returns:
+///     Optional[List[Tuple[gate, equiv_params, equiv_circuit]]]: List of (gate,
+///         equiv_params, equiv_circuit) tuples tuples which, if applied in order
+///         will map from source_basis to target_basis. Returns None if no path
+///         was found.
+pub(crate) fn py_basis_search(
+    py: Python,
+    equiv_lib: &mut EquivalenceLibrary,
+    source_basis: HashSet<GateIdentifier>,
+    target_basis: HashSet<String>,
+) -> PyObject {
+    basis_search(
+        equiv_lib,
+        source_basis
+            .iter()
+            .map(|(name, num_qubits)| (name.as_str(), *num_qubits))
+            .collect(),
+        target_basis.iter().map(|name| name.as_str()).collect(),
+    )
+    .into_py(py)
+}
+
+type BasisTransforms = Vec<(GateIdentifier, BasisTransformIn)>;
+/// Search for a set of transformations from source_basis to target_basis.
+///
+/// Performs a Dijkstra search algorithm on the `EquivalenceLibrary`'s core graph
+/// to rate and classify different possible equivalent circuits to the provided gates.
+///
+/// This is done by connecting all the nodes represented in the `target_basis` to a dummy
+/// node, and then traversing the graph until all the nodes described in the `source
+/// basis` are reached.
+pub(crate) fn basis_search(
+    equiv_lib: &mut EquivalenceLibrary,
+    source_basis: HashSet<(&str, u32)>,
+    target_basis: HashSet<&str>,
+) -> Option<BasisTransforms> {
+    // Build the visitor attributes:
+    let mut num_gates_remaining_for_rule: HashMap<usize, usize> = HashMap::default();
+    let predecessors: RefCell<HashMap<GateIdentifier, Equivalence>> =
+        RefCell::new(HashMap::default());
+    let opt_cost_map: RefCell<HashMap<GateIdentifier, u32>> = RefCell::new(HashMap::default());
+    let mut basis_transforms: Vec<(GateIdentifier, BasisTransformIn)> = vec![];
+
+    // Initialize visitor attributes:
+    initialize_num_gates_remain_for_rule(equiv_lib.graph(), &mut num_gates_remaining_for_rule);
+
+    // TODO: Logs
```

Comment:
You can just remove this we don't need to bother with pylogging here. The basis search debug log messages weren't really ever useful, even for debugging the basis translator.

```suggestion
```